cluster/images/etcd-version-monitor/etcd-version-monitor.go

```diff
@@ -0,0 +1,275 @@
+func getGRPCRequestCount() {
+	// Map for storing last obtained request count for a given grpc request type.
+	lastCount := make(map[GRPCLabels]float64)
+
+	for {
+		// Prepare the prom2json and jq commands to be executed in a pipe.
+		cmd1 := exec.Command("/bin/prom2json", etcdMetricsScrapeURI)
+		cmd2 := exec.Command("jq", ".[]|select(.name==\"etcd_grpc_requests_total\").metrics")
+		r, w := io.Pipe()
+		cmd1.Stdout = w
+		cmd2.Stdin = r
+		cmd2Out, _ := cmd2.StdoutPipe()
+
+		// Run the prom2json command.
+		if err := cmd1.Start(); err != nil {
+			log.Fatal("Failed to start prom2json command: ", err)
+			time.Sleep(scrapeTimeout)
+			continue
+		}
+		// Run the jq command.
+		if err := cmd2.Start(); err != nil {
+			log.Fatal("Failed to start jq command: ", err)
+			time.Sleep(scrapeTimeout)
+			continue
+		}
+		// Wait for prom2json to finish and close w.
+		if err := cmd1.Wait(); err != nil {
+			log.Fatal("Failed to finish prom2json command: ", err)
+			time.Sleep(scrapeTimeout)
+			continue
+		}
+		w.Close()
+		// Read output from jq command and wait for it to finish.
+		out, _ := ioutil.ReadAll(cmd2Out)
+		if err := cmd2.Wait(); err != nil {
+			log.Fatal("Failed to finish jq command: ", err)
+			time.Sleep(scrapeTimeout)
+			continue
+		}
+
+		// Parse the grpc request counts from the json response.
+		var requestCounts []EtcdGRPCRequestCount
+		if err := json.Unmarshal(out, &requestCounts); err != nil {
+			log.Fatal("JSON decoder for grpc request count: ", err)
+			time.Sleep(scrapeTimeout)
+			continue
+		}
+
+		// Increment the request count metrics based on current and previous value.
+		for _ , requestCount := range requestCounts {
+			requestType := requestCount.GrpcLabels
+			newCount, _ := strconv.ParseFloat(requestCount.Count, 64)
+			oldCount := 0.0
+			if val, ok := lastCount[requestType]; ok {
+				oldCount = val	
+			}
+			lastCount[requestType] = newCount
+			if newCount > oldCount {
+				etcdGRPCRequestsTotal.With(prometheus.Labels{
+					"grpc_method": requestType.GrpcMethod,
+					"grpc_service": requestType.GrpcService,
+				}).Add(newCount - oldCount)
+			}
+		}
+		time.Sleep(scrapeTimeout)	
+	}
+}
+
+func main() {
+	// Register the commandline flags passed to the tool.
+	registerFlags(pflag.CommandLine)
+	pflag.Parse()
+
+	// Register the metrics we defined above with prometheus.
+	prometheus.MustRegister(etcdVersionFetchCount)
+	prometheus.MustRegister(etcdVersionedDataSize)
+	prometheus.MustRegister(etcdGRPCRequestsTotal)
+	prometheus.Unregister(prometheus.NewGoCollector())	// Unregister go metrics.
+
+	// Spawn threads for periodically scraping etcd version metrics and populating our metrics.
+	go getVersion()
+	go getVersionedDataSize()
+	go getGRPCRequestCount()
+
+	// Serve our metrics on listenAddress/metricsPath.
+	log.Infoln("Listening on: ", listenAddress)
+	http.Handle(metricsPath, prometheus.Handler())
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I don't think this is useful.

It would be much more useful to e.g. export when is the last time we updated each of the metrics.